cmd/internal/planetscale_edge_database.go

```diff
@@ -39,9 +40,25 @@ type PlanetScaleEdgeDatabase struct {
 }
 
 func (p PlanetScaleEdgeDatabase) CanConnect(ctx context.Context, psc PlanetScaleSource) error {
+	if err := p.isEdgePassword(ctx, psc); err != nil {
+		return errors.New("Unable to reach connect enabled query endpoint for PlanetScale. Please ensure that your organization is enrolled in the Connect beta.")
+	}
+
 	return p.Mysql.PingContext(ctx, psc)
 }
 
+func (p PlanetScaleEdgeDatabase) isEdgePassword(ctx context.Context, psc PlanetScaleSource) error {
```

Comment:
```suggestion
func (p PlanetScaleEdgeDatabase) checkEdgePassword(ctx context.Context, psc PlanetScaleSource) error {
```

This will return an error if it is not an edge password, so we should call the method something like `check` instead of `is`